packages/flutter_tools/lib/src/reporting/usage.dart

```diff
@@ -187,13 +214,21 @@ class _DefaultUsage implements Usage {
     if (usingLogFile) {
       _analytics = LogToFileAnalytics(logFilePath);
     } else {
-      _analytics = AnalyticsIO(
-            _kFlutterUA,
-            settingsName,
-            version,
-            documentDirectory:
-                configDirOverride != null ? globals.fs.directory(configDirOverride) : null,
-          );
+      try {
+        _analytics = analyticsIOFactory(
+          _kFlutterUA,
+          settingsName,
+          version,
+          documentDirectory: configDirOverride != null
+            ? globals.fs.directory(configDirOverride)
+            : null,
+        );
+      } on Exception catch (e) {
+        globals.printError('Failed to initialize analytics reporting: $e');
```

Comment:
Is this an actionable message in most cases? i.e. if this is due to running on a machine with restricted rw permissions, I could see this just being spam.

Reply:
Sorry, with intentionally restricted rw permissions

Reply:
Changed to printTrace.